docs/source/torch_nccl_environment_variables.rst

```diff
@@ -0,0 +1,33 @@
+.. _torch_nccl_environment_variables:
+
+PYTORCH NCCL Environment Variables
+==================================
+.. list-table::
+  :header-rows: 1
```

Comment:
hmm, do we want to mention default values here? pro: i think its helpful. con: its gonna be out of date. 

Maybe give a pointer instead to the the code where the default values are set?